templates/packer/ansible/roles/common/files/worker-initialization-scripts/worker_executor.sh

```diff
@@ -1,19 +1,31 @@
 #!/bin/bash
 set -euo pipefail
 
-# Don't subscribe on fedora
 source /etc/os-release
+source /tmp/cloud_init_vars
+
+# Don't subscribe on fedora
 if [ "$ID" != fedora ]; then
   /usr/local/bin/aws secretsmanager get-secret-value \
     --secret-id executor-subscription-manager-command | jq -r ".SecretString" > /tmp/subscription_manager_command.json
   jq -r ".subscription_manager_command" /tmp/subscription_manager_command.json | bash
   rm -f /tmp/subscription_manager_command.json
 fi
 
+echo "Writing osbuild_executor config to worker configuration."
+OSBUILD_EXECUTOR_IAM_PROFILE=${OSBUILD_EXECUTOR_IAM_PROFILE:-osbuild-executor}
+sudo tee -a /etc/osbuild-worker/osbuild-worker.toml > /dev/null << EOF
+[osbuild_executor]
+type = "aws.ec2"
+iam_profile = "${OSBUILD_EXECUTOR_IAM_PROFILE}"
+EOF
```

Comment:
Ah this needs to be configured on the worker, not the executor, sorry for not being clear.

Reply:
ACK, I had it outside, but then wanted to optimize things and make changes only to existing files :facepalm: